src/Get.tsx

```diff
@@ -225,7 +225,27 @@ class ContextlessGet<TData, TError> extends React.Component<
       return null;
     }
 
-    this.setState({ loading: false, data: resolve!(data) });
+    let resolvedData: TData | null = null;
+    let resolveError: GetDataError<TError> | null = null;
+
+    try {
+      if (resolve) {
+        const resolvedDataOrPromise: TData | Promise<TData> = resolve(data);
+        resolvedData = (resolvedDataOrPromise as { then?: any }).then
```

Comment:
What is the reasoning behind this type assertion? Tbh it looks a little messy.

Reply:
I wasn't able to get it to type-check any other way - looks like detecting whether something is a promise or a function is not that straightforward to follow by the compiler. Do you have any idea how this could improve?

Reply:
I think we can pair up on it. I'd love to understand the complexity more.

Reply:
Awesome - I managed to confirm via the parcel example that the new features are working, now it's just a matter of cleaning up the code.